.github/workflows/rtools40.yml

```diff
@@ -15,6 +15,12 @@ jobs:
           { msystem: ucrt64, toolchain: "ucrt-x86_64" }
         ]
       fail-fast: false
+    env:
```

Comment:
Is this change necessary?

Reply:
No not necessary at this time.  
(Was trying to make more consistent between rtools40/msys2, and addressed deprecated-only use in rtools40 of MINGW_INSTALLS)

Reply:
> * Two high level requirements for changes:
>       
>       * Unit tests must use separately-timestamped writes.
tiledb_unit tests are using separately-timestamped writes
unit_blob_array tests were failing with separately-timestamped writes thus were removed (trying to move along), but will revisit, (may have been a problem with my implementation there)

>       * Timestamp handling must be moved to a `BlobArray::open` implementation as previously discussed.

(Some further discussion with Seth regarding this since our joint conversation)
Due to (intentional) design of API allowing 'other' use of array, BlobArray open() was deemed an inappropriate place to set timestamps as there is no easy way to reliably restore any in place by client/user, they may have 'set' a timestamp_end_, and further opened somewhere to result in an opened_at_timestamp_end_ (which they have no direct influence over), and which continues to be used in any subsequent open()s.
The synchronization Seth has spec'd requires that both timestamp_end_ and timestamp_end_opened_at_ be set (and to same value).
The actual single routine where the write's are performed -could- however restore any previous values (whether success or error occurred) using a ScopedExecutor and hence timestamps are currently being set there and restored there.
(This does not recover everything, due to current 'flow' of various actions, I found no way for the (pre-)generated metadata uri to be restored, but at least the timestamps involved are restored which would avoid silently affecting other 'outside' opens.  Since 'external' API users can't currently directly affect the timestamp_end_opened_at_ used for uri generation, I don't think this forced metadata uri generation with 'local' timestamp is a problem.)